### PR TITLE
migrate membership-checker to be extensible

### PR DESCRIPTION
**Pull request by ScottSuarez**
Migrate membership checker to be extensible and run directly. 

Switching to a COBRA CLI model. Plan is to migrate more and more CI functionality into this package over time.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by shuyama1 — COMMENTED** [review]
LGTM overall! Nice work!
As discussed offline, two things may still be needed:
1. fetch the main instead of base branch for membership-checker to get the latest version of the reviewer list.
2. update the unit tests run workflow

**Review by shuyama1 — APPROVED** [review]

File: .ci/gcb-community-checker.yml [hunk 1]
```diff
@@ -1,23 +1,71 @@
...
+
+  # Fetch main (only if it's not already present)
+  - name: "gcr.io/cloud-builders/git"
+    args:
+      - fetch
+      - origin
+      - main
+
+  # Display commit log for clarity
+  - name: "gcr.io/cloud-builders/git"
```
[the remark this answers is not in the document]
Comment (ScottSuarez) [reply .ci/gcb-community-checker.yml:33]: ahh it can likely be removed. Let me do that.

[the remark this answers is not in the document]
Comment (ScottSuarez) [reply .ci/gcb-community-checker.yml:33]: Leaving for now, can remove it later.

**Comment by shuyama1** [comment]
Let's make sure the vacation list is in sync with https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/containers/membership-checker/membership.go#L32 before merging this PR

File: .ci/gcb-community-checker.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (shuyama1) [reply .ci/gcb-community-checker.yml:33]: Sounds goode

**Pull request merged**